guide_drafts/schema-in-depth.md

```diff
@@ -0,0 +1,218 @@
+Schema in Depth
+===============
+
+In this guide we're going to look at what exactly `infer_schema!`, `infer_table_from_schema!`, and `table!` do. For `table!` we will show a simplified version of the actual code that gets generated, and explain how each piece is relevant to you. If you've ever been confused about what exactly is getting generated, or what `use schema::posts::dsl::*` means, this is the right place to be.
+
+`infer_schema!` is a macro provided by `diesel_codegen` when you have enabled the feature for one or more database backends. The macro will establish a database connection at compile time, query for a list of all the tables, and generate `infer_table_from_schema!` for each one. `infer_schema` will skip any table names which start with `__`.
```

Comment:
I'm not sure I understand what your first correction means? The point I was trying to make was that the macro is only present if you've enabled at least one backend (otherwise it's just the derives)